Profiler/Basics/GridProfiler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Profiler.Core;
+using Profiler.Util;
+using Sandbox.Game.Entities;
+using VRage.Game.Entity;
+
+namespace Profiler.Basics
+{
+    public sealed class GridProfiler : BaseProfiler<MyCubeGrid>
+    {
+        readonly GameEntityMask _mask;
+        readonly Action<MyEntity> _onGameEntityRemoved;
+
+        public GridProfiler(GameEntityMask mask)
+        {
+            _mask = mask;
+
+            _onGameEntityRemoved = gameEntity =>
+            {
+                if (!(gameEntity is MyCubeGrid grid)) return;
+                RemoveEntry(grid);
+            };
+
+            MyEntities.OnEntityRemove += _onGameEntityRemoved;
```

Comment:
Since these profilers are short lived you can skip removing the profiler entries when their entity is deleted.